### DISCUSSION TITLE
Amiga Emulation help

### DISCUSSION THREAD

**Discussion by Allismeaningless** [OnionUI/Onion#1119] [Q&A] [answered]
Hi all,

First post here. Just got my Miyoo Mini Plus and installed onion. It's fantastic and many thanks to all who have made this project work.

I have a few questions on Amiga Emulation. I have got all my roms and kick-starts sorted and some of the games work really well. But there are a few issues:

1. Load times are the same as on the Amiga, which is apocalyptic. 
2. Games with multiple disks don't swap.
3. Quite a few games just run badly.

I think the solution to some of these is, as on the Amiga 500 mini, the WHDLoad approach. However I am not technical so have no idea if this is something that is or could be implemented in the Miyoo.

If anyone knows any solutions to having Amiga games run well (ok those needing a keyboard and mouse are stuffed but most will be good). Perhaps it's a selected ROM set or some settings. Any advice would be appreciated thanks.

**Comment by Allismeaningless** [comment]
I have attempted to get somewhere with this on my own. There is an Amiga WHD Games pack on internet archive where the LHA files are converted to ZIP files. 

This, surprisingly, works out of the box on the Miyoo. I also managed to scrape about 70% of the game images (which is pretty decent considering there are multiple versions of a lot of the games).

I have only tested on a few games so far. Alien Breed (SE) now runs where it didn't before. Alien Breed 2 now runs without graphical glitches. Load times appear to be better, though I need to try a few other games.

Apologies this has very little to do with Onion OS other than its the system being used. Hopefully it helps the more knowledgeable think of some ways to improve this process.

**Comment by ghost** [comment, marked as answer]
I just use LHA images from GamesNostalgia (didn't need to convert to zip etc.) and they work perfectly fine when prepared correctly - maybe that game pack you mention is not up to par or just outdated - for many games those packages get constant updates. 
For ADF images you should be able to change disks in the retroarch menu (I didn't check this as I moved completely to LHA images, but it worked fine on Garlic OS, so it should here too I guess). It's a little cumbersome though, WHDLoad images take care of that thankfully.

**Reply by Allismeaningless** [reply]
Cool thanks for the tip. I am currently curating a collection of ones that work between the WHDLoad and the no intro packs. But will go to that website for updated versions. Once I have them all collected and scraped I may do an upload for prosperity. 

Most of the WHDLoad games work well, but there is the occasional crash out to DOS with a WHDLoad related error, CPU requirement or so on.  Dynamite dux for example works on the no intro set but not the WHDLoad set.
